drivers/shared/executor/utils.go

```diff
@@ -84,7 +88,14 @@ func ReattachToExecutor(reattachConfig *plugin.ReattachConfig, logger hclog.Logg
 		AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
 		Logger:           logger.Named("executor"),
 	}
-	return newExecutorClient(config, logger)
+	exec, pluginClient, err := newExecutorClient(config, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+	if _, err := exec.Version(); err != nil {
```

Comment:
Is reattach always done before new executors are launched? Otherwise, is it possible that the process that is listening on the previous port is just a different executor and the Version call succeeds despite the PID not being an executor? Would it make more sense to check process name or creation time to ensure we are re-attaching the correct process?

Reply:
Great. Thank you for clarifying.